compiler/semmagic.nim

```diff
@@ -13,6 +13,8 @@
 proc semAddr(c: PContext; n: PNode; isUnsafeAddr=false): PNode =
   result = newNodeI(nkAddr, n.info)
   let x = semExprWithType(c, n)
+  if x.kind == nkSym:
```

Comment:
I don't think that's true. sfAddrTaken would force the "s" symbol to be initialized and used indirectly: `var s = [newArray(....)]`. But such indirection is not needed, unless we want to take address of the whole array, whatever that means.